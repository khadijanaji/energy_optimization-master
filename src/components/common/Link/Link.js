// @flow

import * as React from "react";

export type LinkPropsType = {
    children: React.Node,
    url: string,
    navigateToUrlOnClick: Function
};

const Link = ({
  children,
  url,
  navigateToUrlOnClick
}: LinkPropsType) => (<div onClick={navigateToUrlOnClick(url)}>{children}</div>);

export default Link;
