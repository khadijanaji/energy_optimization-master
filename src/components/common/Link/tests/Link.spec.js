/* globals describe it expect*/

import React from "react";
import {shallow} from "enzyme";
import Link from "../Link";

const navigateToUrlOnClick = jest.fn();

describe("Component", () => {
  describe("Link", () => {
    it("With url: Should match snapshot", () => {
      const wrapper = shallow(<Link
        navigateToUrlOnClick={navigateToUrlOnClick}
        url={"url"}
      ><div /></Link>);
      expect(wrapper).toMatchSnapshot();
    });

    it("When clicked: Should call navigateToUrlOnClick", () => {
      const wrapper = shallow(<Link
        navigateToUrlOnClick={navigateToUrlOnClick}
        url={"url"}
      ><span /></Link>);
      wrapper.find("div").simulate("click");
      expect(navigateToUrlOnClick).toBeCalled();
    });
  });
});
