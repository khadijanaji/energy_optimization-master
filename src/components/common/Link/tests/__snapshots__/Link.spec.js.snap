// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Link With url: Should match snapshot 1`] = `
<div>
  <div />
</div>
`;
